mux.c

```diff
@@ -9,14 +9,42 @@
 #include "cpu6.h"
 #include "mux.h"
 #include "scheduler.h"
+#include "trace.h"
+
+#define TRACE_WITH_CHAR(val, ...)					\
+	if (trace) {							\
+		fprintf(stderr, "%04X ", cpu6_pc());			\
+		fprintf(stderr, __VA_ARGS__);				\
+		fprintf(stderr, " %x", val);				\
+		if ((val&0x7f) >= 0x20 && val != 0x7F && val != 0xff)	\
+			fprintf(stderr, " ('%c')", val & 0x7f);		\
+		fputc('\n', stderr);					\
+	}
 
 struct MuxUnit mux[NUM_MUX_UNITS];
- // 0 disables interrupts
- // LOAD writes 0 to F20A before transferring execution to a new binary
-static unsigned char rx_ipl_request = 0;
-static unsigned char tx_ipl_request = 0;
-static int irq_cause = -1;
-static uint32_t poll_count = 0;
+static unsigned char irq_level;
+static unsigned char irq_enabled;
+static int irq_cause;
+static uint32_t poll_count;
+
+static void mux_reset(void)
+{
+	int i;
+
+	for (i = 0; i < NUM_MUX_UNITS; i++) {
+		mux[i].status        = MUX_TX_READY;
+		mux[i].lastc         = 0xFF;
+		mux[i].baud          = 9600;
```

Comment:
Frankly speaking i don't know; i didn't check those counters in the schematic. Well, i guess they reset to 0, so it could stand maybe for 1:1 = 19200.
 I didn't spend time on this because we currently don't support real serial ports, so baud rate is irrelevant.